cmd/compute-domain-controller/resourceclaimtemplate.go

```diff
@@ -0,0 +1,356 @@
+	tmpl, err := template.ParseFiles(templatePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse template file: %w", err)
+	}
+
+	var resourceClaimTemplateYaml bytes.Buffer
+	if err := tmpl.Execute(&resourceClaimTemplateYaml, templateData); err != nil {
+		return nil, fmt.Errorf("failed to execute template: %w", err)
+	}
+
+	var unstructuredObj unstructured.Unstructured
+	err = yaml.Unmarshal(resourceClaimTemplateYaml.Bytes(), &unstructuredObj)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal yaml: %w", err)
+	}
+
+	var resourceClaimTemplate resourceapi.ResourceClaimTemplate
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.UnstructuredContent(), &resourceClaimTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert unstructured data to typed object: %w", err)
+	}
+
+	rct, err := m.config.clientsets.Core.ResourceV1beta1().ResourceClaimTemplates(resourceClaimTemplate.Namespace).Create(ctx, &resourceClaimTemplate, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error creating ResourceClaimTemplate: %w", err)
+	}
+
+	return rct, nil
+}
+
+func (m *BaseResourceClaimTemplateManager) Delete(ctx context.Context, cdUID string) error {
+	rcts, err := getByComputeDomainUID[*resourceapi.ResourceClaimTemplate](ctx, m.informer, cdUID)
+	if err != nil {
+		return fmt.Errorf("error retrieving ResourceClaimTemplate: %w", err)
+	}
+	if len(rcts) > 1 {
+		return fmt.Errorf("more than one ResourceClaimTemplate found with same ComputeDomain UID")
+	}
+	if len(rcts) == 0 {
+		return nil
+	}
+
+	rct := rcts[0]
+
+	if rct.GetDeletionTimestamp() != nil {
+		return nil
+	}
+
+	err = m.config.clientsets.Core.ResourceV1beta1().ResourceClaimTemplates(rct.Namespace).Delete(ctx, rct.Name, metav1.DeleteOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("erroring deleting ResourceClaimTemplate: %w", err)
+	}
+
+	return nil
+}
+
+func (m *BaseResourceClaimTemplateManager) RemoveFinalizer(ctx context.Context, cdUID string) error {
+	rcts, err := getByComputeDomainUID[*resourceapi.ResourceClaimTemplate](ctx, m.informer, cdUID)
+	if err != nil {
+		return fmt.Errorf("error retrieving ResourceClaimTemplate: %w", err)
+	}
+	if len(rcts) > 1 {
+		return fmt.Errorf("more than one ResourceClaimTemplate found with same ComputeDomain UID")
+	}
+	if len(rcts) == 0 {
+		return nil
+	}
+
+	rct := rcts[0]
+
+	if rct.GetDeletionTimestamp() == nil {
+		return fmt.Errorf("attempting to remove finalizer before ResourceClaimTemplate marked for deletion")
+	}
+
+	newRCT := rct.DeepCopy()
+	newRCT.Finalizers = []string{}
+	for _, f := range rct.Finalizers {
+		if f != computeDomainFinalizer {
+			newRCT.Finalizers = append(newRCT.Finalizers, f)
+		}
+	}
+	if len(rct.Finalizers) == len(newRCT.Finalizers) {
+		return nil
```

Comment:
note to self: this function removes "our" finalizer, the computeDomainFinalizer. All other finalizers need to be retained.